src/Pages/Applications/DevtronApps/Details/AppConfigurations/Navigation/EnvironmentOverrideRouter.tsx

```diff
@@ -420,25 +368,32 @@ const EnvironmentOverrideRouter = ({
     }
 
     const renderEnvsNav = (): JSX.Element => {
-        if ((isJobView ? jobEnvs : allEnvs).length) {
+        if ((isJobView ? jobEnvs : environments).length) {
             return (
                 <div className="w-100" style={{ height: 'calc(100% - 60px)' }} data-testid="env-override-list">
-                    {(isJobView ? jobEnvs : allEnvs).map((env) => {
-                        return (
-                            !env.deploymentAppDeleteRequest && (
-                                <EnvOverrideRoute
-                                    envOverride={env}
-                                    key={env.environmentName}
-                                    isJobView={isJobView}
-                                    ciPipelines={ciPipelines}
-                                    reload={reloadEnvData}
-                                    appId={appId}
-                                    workflowsRes={workflowsRes}
-                                    isEnvProtected={env.isProtected}
-                                />
-                            )
-                        )
-                    })}
+                    {isJobView
+                        ? jobEnvs.map(
+                              (env) =>
+                                  !env.deploymentAppDeleteRequest && (
+                                      <JobEnvOverrideRoute
+                                          key={env.environmentName}
+                                          envOverride={env}
+                                          ciPipelines={ciPipelines}
+                                          reload={reloadEnvData}
+                                          isEnvProtected={env.isProtected}
+                                      />
+                                  ),
+                          )
+                        : environments.map((env) =>
+                              renderNavItem(
+                                  {
+                                      title: env.environmentName,
+                                      isProtectionAllowed: env.isProtected,
+                                      href: `${URLS.APP_ENV_OVERRIDE_CONFIG}/${env.environmentId}/deployment-template`,
+                                  },
+                                  env.isProtected,
+                              ),
+                          )}
```

Comment:
- Potentially need to add a key by wrapping with fragment
- Use enum for deployment-tempalte